streaming/src/main/scala/org/apache/spark/streaming/util/WriteAheadLogUtils.scala

```diff
@@ -39,6 +39,7 @@ private[streaming] object WriteAheadLogUtils extends Logging {
 
   val DEFAULT_ROLLING_INTERVAL_SECS = 60
   val DEFAULT_MAX_FAILURES = 3
+  val WAL_CLOSE_AFTER_WRITE = "spark.streaming.writeAheadLog.closeAfterWrite"
```

Comment:
Take a look at how rollingIntervalSecs is implemented. That also has
separate configurations for driver and receiver


On Mon, Oct 26, 2015 at 7:47 PM, Burak Yavuz <notifications@github.com>
wrote:

> In
> streaming/src/main/scala/org/apache/spark/streaming/util/WriteAheadLogUtils.scala
> <https://github.com/apache/spark/pull/9285#discussion_r43079268>:
>
> > @@ -39,6 +39,7 @@ private[streaming] object WriteAheadLogUtils extends Logging {
> >
> >    val DEFAULT_ROLLING_INTERVAL_SECS = 60
> >    val DEFAULT_MAX_FAILURES = 3
> > +  val WAL_CLOSE_AFTER_WRITE = "spark.streaming.writeAheadLog.closeAfterWrite"
>
> @tdas <https://github.com/tdas> Noticed that this is going to be super
> hard. How are we going to differentiate between a receiver WAL and a driver
> WAL in the write method of FileBasedWAL?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/9285/files#r43079268>.
>